lib/nodejs/supervisor.ex

```diff
@@ -42,7 +43,11 @@ defmodule NodeJS.Supervisor do
       end
     end
 
-    fn -> :poolboy.transaction(@pool_name, func, :infinity) end
+    fn -> :poolboy.transaction(get_pool_name(opts), func, :infinity) end
+  end
+
+  defp get_pool_name(opts) do
+    String.to_atom("pool_" <> to_string(Keyword.get(opts, :name, @default_pool_name)))
```

Comment:
It might be simpler to just explicitly as for `:name` to be an atom and default to `@default_pool_name` when it isn't pass. Also, might want to consider changing `:name` to `:pool_name`. This would make it so that `:name` is more specific to how the name registration works for the GenServer/Supervisor https://hexdocs.pm/elixir/GenServer.html#module-name-registration which appears useful if you are going to run multiple of these.

Reply:
Thanks for the review!

Every supervisor will need a uniquely named pool, which is why I generated it dynamically from the `:name`.  To clarify, are you are suggesting that I make both `:name` and `:pool_name` parameters that can be passed into the Supervisor so that we can use stuff like `{:via, ...}`?

Reply:
Yeah, that would be the approached I'd take. If we really want to look into having it dynamically generated, the option would be to have something look at what's passed in as `:name` and have it generate the pool name for us.

e.g.
```
defp generate_pool_name({:via, name}}, :"#{name}.Pool"
defp generate_pool_name(name), do: :"#{name}.Pool"

// Supervisor
name = Keyword.get(opts, :name, __MODULE__)

// Pool
pool = generate_pool_name(Keyword.get(opts, :pool_name, @default_pool_name))
```